src/topology/uniform_space/cauchy.lean

```diff
@@ -50,41 +53,38 @@ cauchy_downwards cauchy_nhds
   (show principal {a} ≠ ⊥, by simp)
   (pure_le_nhds a)
 
+/-- The common part of the proofs of `le_nhds_of_cauchy_adhp` and
+`sequentially_complete.le_nhds_of_seq_tendsto_nhds`: if for any entourage `s`
+one can choose a set `t ∈ f` of diameter `s` such that it contains a point `y`
+with `(x, y) ∈ s`, then `f` converges to `x`. -/
+lemma le_nhds_of_cauchy_adhp_aux {f : filter α} {x : α}
+  (adhs : ∀ s ∈ 𝓤 α, ∃ t ∈ f, (set.prod t t ⊆ s) ∧ ∃ y, (y ∈ t) ∧ (x, y) ∈ s) :
+  f ≤ 𝓝 x :=
+begin
+  -- Consider a neighborhood `s` of `x`
+  assume s hs,
+  -- Take an entourage twice smaller than `s`
+  rcases comp_mem_uniformity_sets (mem_nhds_uniformity_iff.1 hs) with ⟨U, U_mem, hU⟩,
+  -- Take `t` and `y` provided by `adhs`
```

Comment:
That comment doesn't add anything to the line below. If you want to document that proof you could use `obtain` in its cosmetic form `obtain ⟨t, t_mem, ht, y, hy, hxy⟩ : ∃ ..., from adhs U U_mem`.

Reply:
Expanded comment.